swiftnav/src/coords/ecef.rs

```diff
@@ -0,0 +1,255 @@
+use nalgebra::{ArrayStorage, Vector3};
+use std::ops::{Add, AddAssign, Mul, MulAssign, Sub, SubAssign};
+
+use crate::coords::LLHDegrees;
+
+use super::{AzimuthElevation, Ellipsoid, LLHRadians, NED, WGS84};
+
+/// WGS84 Earth Centered, Earth Fixed (ECEF) Cartesian coordinates (X, Y, Z).
+///
+/// Internally stored as an array of 3 [`f64`] values: x, y, z all in meters
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Default)]
+pub struct ECEF(Vector3<f64>);
+
+impl ECEF {
+    /// Create an [`ECEF`] object from the given X, Y, Z components
+    #[must_use]
+    pub fn new(x: f64, y: f64, z: f64) -> ECEF {
```

Comment:
For each of these, can we use `Self` over the actual type name for each of these new structs across the PR?

Reply:
Sure, I can't ever settle on one. Is this something that's checked with one of clippy's pedantic lints?